src/integrationTest/java/com/hivemq/extensions/cluster/discovery/s3/S3DiscoveryIT.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2018-present HiveMQ GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hivemq.extensions.cluster.discovery.s3;
+
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+import com.hivemq.extensions.cluster.discovery.s3.config.TestS3Config;
+import com.hivemq.extensions.cluster.discovery.s3.metrics.TestS3Metrics;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.slf4j.event.Level;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.hivemq.HiveMQContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.S3;
+
+@Testcontainers
+class S3DiscoveryIT {
+
+    private static final @NotNull String BUCKET_NAME = "hivemq";
+
+    private final @NotNull Network network = Network.newNetwork();
+
+    @Container
+    private final @NotNull LocalStackContainer localstack =
+            new LocalStackContainer(DockerImageName.parse("localstack/localstack").withTag("latest")).withServices(S3)
```

Comment:
We try to avoid `latest` as tag. We should specify this to e.g. `1.2.0`. Same in the other ITs.